platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/signed/SignedStateFileWriter.java

```diff
@@ -135,27 +143,241 @@ public static void writeStateFile(final Path directory, final SignedState signed
+                    "No preconsensus event files meeting specified criteria found to copy. "
+                            + "Minimum generation non-ancient: {}",
+                    minimumGenerationNonAncient);
+        } else {
+            logger.info(
+                    STATE_TO_DISK.getMarker(),
+                    """
+                            Found {} preconsensus event files meeting specified criteria to copy.
+                            Minimum generation non-ancient: {}
+                            First file to copy: {}
+                            Last file to copy: {}
+                            """,
+                    filesToCopy.size(),
+                    minimumGenerationNonAncient,
+                    filesToCopy.get(0).getPath(),
+                    filesToCopy.get(filesToCopy.size() - 1).getPath());
+        }
+
+        return filesToCopy;
+    }
+
+    /**
+     * Gather all PCES files on disk.
+     *
+     * @param selfId          the id of this node
+     * @param platformContext the platform context
+     * @return a list of all PCES files on disk
+     */
+    @NonNull
+    private static List<PreconsensusEventFile> gatherPreconsensusFilesOnDisk(
+            @NonNull final NodeId selfId, @NonNull final PlatformContext platformContext) throws IOException {
+        final List<PreconsensusEventFile> allFiles = new ArrayList<>();
+        final Path preconsensusEventStreamDirectory =
+                PreconsensusEventFileManager.getDatabaseDirectory(platformContext, selfId);
+        try (final Stream<Path> stream = Files.walk(preconsensusEventStreamDirectory)) {
+            stream.filter(Files::isRegularFile).forEach(path -> {
+                try {
+                    allFiles.add(PreconsensusEventFile.of(path));
+                } catch (final IOException e) {
+                    // Ignore, this will get thrown for each file that is not a PCES file
+                }
+            });
+        }
+
+        if (allFiles.isEmpty()) {
+            logger.warn(STATE_TO_DISK.getMarker(), "No preconsensus event files found to copy");
+        } else {
+            logger.info(
+                    STATE_TO_DISK.getMarker(),
+                    """
+                            Found {} preconsensus files on disk.
+                            First file: {}
+                            Last file: {}""",
+                    allFiles.size(),
+                    allFiles.get(0).getPath(),
+                    allFiles.get(allFiles.size() - 1).getPath());
+        }
+
+        return allFiles;
+    }
+
+    /**
+     * Copy a list of preconsensus event files into a directory.
+     *
+     * @param filesToCopy     the files to copy
+     * @param pcesDestination the directory where the files should be copied
+     */
+    private static void copyPreconsensusFileList(
+            @NonNull final List<PreconsensusEventFile> filesToCopy, @NonNull final Path pcesDestination)
+            throws IOException {
```

Comment:
This method doesn't throw an `IOException`

Reply:
fixed